examples/manipulation/grasp_eval.py

```diff
@@ -19,73 +22,234 @@
+    if not checkpoint_files:
+        raise FileNotFoundError(f"No checkpoint files found in {log_dir}")
+
+    last_ckpt = sorted(checkpoint_files)[-1]
+    runner.load(os.path.join(log_dir, last_ckpt))
+    print(f"Loaded RL checkpoint from {last_ckpt}")
+
+    return runner.get_inference_policy(device=gs.device)
+
+
+def load_bc_policy(env, bc_cfg, log_dir):
+    """Load behavior cloning policy."""
+    # Create behavior cloning instance
+    bc_runner = BehaviorCloning(env, bc_cfg, None, device=gs.device)
+
+    # Find the latest checkpoint
+    checkpoint_files = [f for f in os.listdir(log_dir) if re.match(r"checkpoint_\d+\.pt", f)]
+    if not checkpoint_files:
+        raise FileNotFoundError(f"No checkpoint files found in {log_dir}")
+
+    last_ckpt = sorted(checkpoint_files)[-1]
+    last_ckpt_file = os.path.join(log_dir, last_ckpt)
+    print(f"Loaded BC checkpoint from {last_ckpt_file}")
+    bc_runner.load(last_ckpt_file)
+
+    return bc_runner._policy
+
+
+def get_stereo_frame(env, step_count):
+    """Get stereo frame as numpy array."""
+    # Get individual camera images
+    rgb_left, _, _, _ = env.left_cam.render(rgb=True, depth=False)
+    rgb_right, _, _, _ = env.right_cam.render(rgb=True, depth=False)
+
+    # Convert to numpy arrays and format for OpenCV
+    left_img = rgb_left[0].cpu().numpy()  # Take first environment
+    right_img = rgb_right[0].cpu().numpy()  # Take first environment
+
+    # Convert from RGB to BGR for OpenCV
+    left_img = cv2.cvtColor(left_img, cv2.COLOR_RGB2BGR)
+    right_img = cv2.cvtColor(right_img, cv2.COLOR_RGB2BGR)
+
+    # Resize images for better display
+    scale = 2.0
+    left_img = cv2.resize(left_img, (int(left_img.shape[1] * scale), int(left_img.shape[0] * scale)))
+    right_img = cv2.resize(right_img, (int(right_img.shape[1] * scale), int(right_img.shape[0] * scale)))
+    # Concatenate images horizontally (side by side)
+    stereo_img = np.hstack([left_img, right_img])
+    # Add a vertical line separator between the two images
+    separator_x = left_img.shape[1]
+    cv2.line(
+        stereo_img,
+        (separator_x, 0),
+        (separator_x, stereo_img.shape[0]),
+        (255, 255, 255),
+        2,
+    )
+    return stereo_img
+
+
+def display_stereo_images(env, step_count):
+    """Display left and right RGB images side by side in one window."""
+    stereo_img = get_stereo_frame(env, step_count)
+
+    # Display combined image
+    cv2.imshow("Stereo Cameras", stereo_img)
+
+    # Wait for key press (1ms delay)
+    key = cv2.waitKey(1)
+    if key == 27:  # ESC key
+        return False
+    return True
+
+
+def save_frames_as_video(frames, video_path, fps=60):
+    """Save a list of frames as a video."""
+    if not frames:
+        print("No frames to save!")
+        return
+
+    # Get frame dimensions from the first frame
+    height, width = frames[0].shape[:2]
+
+    # Create video writer
+    fourcc = cv2.VideoWriter_fourcc(*"mp4v")
```

Comment:
cv2 is not the best option to record video in python. I would recommend [PyAV](https://github.com/PyAV-Org/PyAV). But it is probably acceptable in this context.

Reply:
good to know the PyAV package. didn't know it before.

Reply:
PyAV is great. It is used in torch and co. under the hood.